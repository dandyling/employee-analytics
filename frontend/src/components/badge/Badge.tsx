import { ReactNode } from "react";

export type ColorType = "warning" | "success" | "normal";

interface Props {
  children: ReactNode;
  className?: string;
  variant?: ColorType;
}

export const Badge = (props: Props) => {
  const { children, className = "", variant = "warning" } = props;

  const color = getVariantColor(variant);

  return (
    <div className={`p-1 rounded-full w-28 ${className} ${color}`}>
      {children}
    </div>
  );
};

export const getVariantColor = (variant: ColorType) => {
  switch (variant) {
    case "success":
      return "bg-success";
    case "warning":
      return "bg-warning";
    case "normal":
    default:
      return "bg-normal";
  }
};
