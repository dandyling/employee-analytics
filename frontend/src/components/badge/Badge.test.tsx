import { render, screen } from "@testing-library/react";
import React from "react";
import { Badge, getVariantColor } from "./Badge";

test("renders string", () => {
  render(<Badge>String</Badge>);
  const childElement = screen.getByText(/String/i);
  expect(childElement).toBeInTheDocument();
});

test("renders number", () => {
  render(<Badge>1.23</Badge>);
  const childElement = screen.getByText(/1.23/i);
  expect(childElement).toBeInTheDocument();
});

test("renders empty child for null", () => {
  const { container } = render(<Badge>{null}</Badge>);
  expect(container.firstChild).toBeEmptyDOMElement();
});

test("renders empty child for undefined", () => {
  const { container } = render(<Badge>{undefined}</Badge>);
  expect(container.firstChild).toBeEmptyDOMElement();
});

test("renders react node", () => {
  render(
    <Badge>
      <div>node</div>
    </Badge>
  );
  const nodeElement = screen.getByText(/node/i);
  expect(nodeElement).toBeInTheDocument();
});

test("renders correctly", () => {
  const wrapper = (
    <Badge>
      <div>node</div>
    </Badge>
  );
  expect(wrapper).toMatchSnapshot();
});

test("returns bg-success variant class correctly", () => {
  expect(getVariantColor("success")).toEqual("bg-success");
});

test("returns bg-warning variant class correctly", () => {
  expect(getVariantColor("warning")).toEqual("bg-warning");
});

test("returns bg-normal variant class correctly", () => {
  expect(getVariantColor("normal")).toEqual("bg-normal");
});
