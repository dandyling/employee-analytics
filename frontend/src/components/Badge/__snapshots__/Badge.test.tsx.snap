// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Badge>
  <div>
    node
  </div>
</Badge>
`;
